system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -9,25 +9,33 @@ import {TOP_SITES_MAX_SITES_PER_ROW} from "common/Reducers.jsm";
 import {TopSiteForm} from "./TopSiteForm";
 import {TopSiteList} from "./TopSite";
 
+function topSiteIconType(link) {
+  if (!link) {
+    return null;
```

Comment:
Be careful about this case, cause the following `acc[topSiteIconType(link)]++` might generate some funky results.

Reply:
There is actually a mistake here. That shouldn't be in. There is always a link object being passed in.